src/main/flight/autopilot.c

```diff
@@ -91,14 +190,162 @@ void altitudeControl(float targetAltitudeCm, float taskIntervalS, float vertical
     DEBUG_SET(DEBUG_AUTOPILOT_ALTITUDE, 7, lrintf(altitudeF));
 }
 
-bool isBelowLandingAltitude(void)
+void setSticksActiveStatus(bool areSticksActive)
 {
-    return getAltitudeCm() < 100.0f * autopilotConfig()->landing_altitude_m;
+    posHold.sticksActive = areSticksActive;
+}
+
+void setTargetLocation(gpsLocation_t newTargetLocation) {
+    currentTargetLocation = newTargetLocation;
+}
+
+bool positionControl(void) {
+
+    // exit if we don't have a GPS 3D fix
+    if (!STATE(GPS_FIX)) {
+        return false; // cannot proceed without a GPS location
+    }
+    // and if no valid heading from GPS, or no mag
+    if (!canUseGPSHeading
+#ifdef USE_MAG
+        && !compassIsHealthy()
+#endif
+        ) {
+        return false;
+    }
+    
+    if (!wasThrottleRaised()) {
+        return false;
+    }
+
+    if (isNewGPSDataAvailable()) {
+        posHold.gpsDataIntervalS = getGpsDataIntervalSeconds(); // interval for current GPS data value 0.01s to 1.0s
+        posHold.gpsDataFreqHz = 1.0f / posHold.gpsDataIntervalS;
+        if (posHold.sticksActive) {
+            // if a Position Hold deadband is set, and sticks are outside deadband, allow pilot control in angle mode
+            resetPositionControlParams(&posHold.direction[NORTH_SOUTH]);
+            resetPositionControlParams(&posHold.direction[EAST_WEST]);
+            posHold.pidSum[AI_ROLL] = 0.0f;
+            posHold.pidSum[AI_PITCH] = 0.0f;
+        } else {
+            // first get xy distances from current location (gpsSol.llh) to target location
+            vector2_t gpsDistance;
+            GPS_distances(&gpsSol.llh, &currentTargetLocation, &gpsDistance.y, &gpsDistance.x); // Y is north, X is south
```

Comment:
The comment is wrong

And .. BF uses NEU coordinate system for GPS. It is quite problematic (heading is clockwise, opposite to common mathematical angle conventions, axes seem to be swapped (When you rotate X to the right, Y is pointing down instead of up). But mixing NEU with ENU will lead to even more confusion and problems.

And converting BF to ENU is problematic too - heading will be inverted AND offset by 90 deg :-(

So, IMO, GPS_distance should return vector2_t *, `<dLat, dLon>`. And this code shall accept it as a fact.. (I hate this situation)

Reply:
Isn't NED the most common coordinate system?

Reply:
https://en.wikipedia.org/wiki/Axes_conventions#:~:text=World%20reference%20frames%3A%20ENU%20and%20NED,-Main%20article%3A%20Local&text=Basically%2C%20as%20lab%20frame%20or,NED)%2C%20used%20specially%20in%20aerospace
Its probably worth the effort to convert to a NED coordinate system. A NEU system is very nontypical.

Reply:
@Quick-Flash : Minor problem with NED is that is is usually centered on plane, with ground offset being positive. But BF mostly uses origin on ground/home, so altitude will be negative.

NEU is heritage from old times. GPS coordinates were taken as X,Y / Z was altitude; heading used normal 'compass' meaning. And equations were multiplied by +-1 until 'it worked'.
It was so real pain deciphering IMU calculations..
Magic (and IMO incorrect) conversions in SITL are another consequence of this.

I'll be glad to help you, and there are few other contributors that can help with conversion. But I don't expect it to be smooth.
Maybe it would be best to document current reference frames (https://github.com/betaflight/betaflight/discussions/11903) and then decide how to proceed ...

Reply:
At least ENU or NED should be picked. I don't think that NEU should be used.

> I'll be glad to help you, and there are few other contributors that can help with conversion. But I don't expect it to be smooth. Maybe it would be best to document current reference frames (#11903) and then decide how to proceed ...

Yeah, but I mean these few that could contribute are the few that will be bit in the butt if this doesn't get corrected sooner than later.

Reply:
This conversation explains why I had so much trouble getting the controller 'signs' right.  Each rotation could reverse the sign of the input, and the basic distance, or 'error', is just a positive or negative distance from the target point. 
I'm happy for you guys to properly name all these things after the PR is merged, but at the moment I want to keep the simple north-south / east-west identifiers for earth referenced values, and pitch / roll for quad based values.  It's just a lot easier for me to follow the code that way.

Reply:
@ctzsnooze: Problem is that you spent a lot of time making sense of BF coordinated and ended up with another version of coordinate system. Anyone working of this code will have do the same (and probably will give up).
And if you changed sign somewhere 'because it works that way', it is almost impossible to understand the code without fixing actual reason why it was inverted.
It is acceptable as WIP, but merging it will add huge burden to anyone trying to work on BF.

This already happened to Mahony IMU (and quaternions). Getting compass and CoG work took considerable effort.
A lot of people (including myself) resorted to 'add minus here or here to make it work' when working in IMU, making the problem even worse.

Reply:
I’m happy to set it up however it should be. I just didn’t know, when I wrote it, what the conventions were, and it wasn’t documented anywhere that I could find in the code. 

I’m fairly sure that in earth frame, I made it so that if the craft was North of the target, that was a positive distance. Is that right?

If so, what should the sign be when the craft is East of the target?

Then we should do the maths so that the PIDsum in the earth frame has the opposite sign.  

And then we need to know the convention for pitch and roll. For that I think that positive pitch means nose down. 

That being the case, if the craft faces North, and is North of target, the distance should be positive, the pidSum  should be negative, and the quad should pitch back due to negative pitch applied. 

If I can be given clear instructions for Roll then we can make it however it should be.

Reply:
Situation in current PR in a bit problematic.
In pid, you get error = target - current. GPS_distances does this, but with argument order reversed. It is better to order arguments to match subtraction (`void GPS_distances(const gpsLocation_t *to, const gpsLocation_t *from, ... `), It is also a bit more logical, you are asking about 'to' location, 'from' is a bit implicit here.
But GPS_distance_cm_bearing is already 'wrong' ... 

IMO error should be positive when target is further north relative to drone (and to the east from drone)

> If so, what should the sign be when the craft is East of the target?

IMO negative


>  And then we need to know the convention for pitch and roll. For that I think that positive pitch means nose down.

Yes, flying forward, and with zero heading flying to the north. With heading=90 it flies east.
And with positive Roll, quad flies to the right (and East with zero heading)

> That being the case, if the craft faces North, and is North of target, the distance should be positive, the pidSum should be negative, and the quad should pitch back due to negative pitch applied.

Distance is negative, pidsum is negative, desired pitch is negative (front side up) and it flies backwards, to north

> If I can be given clear instructions for Roll then we can make it however it should be.

if the craft faces North, and is East of target,

Distance is negative, pidsum is negative, desired roll is negative (right side up) and it flies to the left, to west